### ISSUE TITLE
TestCuda.test_streams is flaky on circleCI windows.

### ISSUE THREAD

**Issue by ailzhang** [pytorch/pytorch#32949]
Example failure log: 
https://app.circleci.com/jobs/github/pytorch/pytorch/4406011
```
test_type_conversions (__main__.TestCuda) ... ok

======================================================================
FAIL: test_streams (__main__.TestCuda)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_cuda.py", line 988, in test_streams
    self.assertFalse(default_stream.query())
AssertionError: True is not false

----------------------------------------------------------------------
Ran 98 tests in 44.548s

FAILED (failures=1, skipped=48)
Traceback (most recent call last):
  File "run_test.py", line 457, in <module>
    main()
  File "run_test.py", line 450, in main
    raise RuntimeError(message)
RuntimeError: test_cuda failed!
```

cc @ezyang @gchanan @zou3519

**Comment by ailzhang** [comment]
Found in PR https://github.com/pytorch/pytorch/pull/32945

**Comment by mruberry** [comment]
This should be fixed with #33146.
